src/oop/special_class_method/hard/q9/MainTest.java

```diff
@@ -0,0 +1,210 @@
+package oop.special_class_method.hard.q9;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class MainTest {
+    // Parsons with Distractors
+    private final String className = "Earth";
+    private final String doubleFieldName = "landMass";
+    private final String intFieldName = "population";
+    private final String setIntMethodName = "setPopulation";
+    private final String getIntMethodName = "getPopulation";
+    private final String getDoubleMethodName = "getLandMass";
+    private final String setDoubleMethodName = "setLandMass";
+    private ObjectTest testClass;
+
+    @BeforeEach
+    public void setUp() {
+        String classString = "oop.special_class_method.hard.q9." + className;
+        testClass = new ObjectTest(classString);
+    }
+
+    @Test
+    public void earthClassHasCorrectAttributes() {
+        assertTrue(testClass.hasField(doubleFieldName, double.class),
+                "Your " + className + " class is missing the " + doubleFieldName + " attributes.");
+        assertTrue(testClass.hasModifier(doubleFieldName, "private"),
+                "Your " + doubleFieldName + " does not have the correct visibility modifier.");
+        assertTrue(testClass.hasField(intFieldName, int.class),
+                "Your " + className + " class is missing the " + intFieldName + " attributes.");
+        assertTrue(testClass.hasModifier(intFieldName, "private"),
+                "Your " + intFieldName + " does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, 300),
+                Arguments.of(234.4, 100),
+                Arguments.of(654.45, 365),
+                Arguments.of(43.3, 14)
+
+        );
+    }
+
```

Comment:
There is no assert to check if the constructor exists, only that is initializes attributes to the correct value and then that it has the correct visibility modifier